src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionServiceTest.java

```diff
@@ -204,6 +201,28 @@ public void shouldCorrectlyExtractLevelOfAssurance() {
     }
 
 
+    @Test
+    public void shouldThrowExceptionWhenLevelOfAssuranceNotPresent() {
+        Assertion authnAssertion = anAuthnStatementAssertion(null, "requestId").buildUnencrypted();
+        Assertion mdsAssertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+
+        exception.expect(SamlResponseValidationException.class);
+        exception.expectMessage("Expected a level of assurance.");
+        nonMatchingAssertionService.translateSuccessResponse(ImmutableList.of(authnAssertion, mdsAssertion), "requestId", LEVEL_2, "default-entity-id");
```

Comment:
If we provide the `LevelOfAssuranceValidator` in the constructor, we can provide a mock and `verify` on that mock here.

Reply:
Implemented the mock use